heads/src/test/java/com/dumbdogdiner/stickyapi/mockedplugin/StickyAPIPlugin.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.mockedplugin;
+
+
+import com.dumbdogdiner.stickyapi.bukkit.util.StartupUtil;
+import com.dumbdogdiner.stickyapi.common.translation.LocaleProvider;
+import lombok.Getter;
+import org.bukkit.plugin.PluginDescriptionFile;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.bukkit.plugin.java.JavaPluginLoader;
+
+import java.io.File;
+
+public class StickyAPIPlugin extends JavaPlugin {
```

Comment:
Cloned from the previous review: 
> Did we ever agree on how we would implement this? Thought for quite a long time it makes much more sense to just pass plugin-specific data about. Do correct me if I'm wrong.

Reply:
I don't remember us agreeing, and I also don't remember an easy way to do testing on plugin type functions without making a mocked plugin. please correct me

Reply:
Okay, can you refactor this out into a separate branch and then we can discuss how it would be implemented from there? Would be highly appreciated <3

Reply:
Its needed for testing heads, I can refactor but it's a pain due to testing. @Prouser123 thoughts?